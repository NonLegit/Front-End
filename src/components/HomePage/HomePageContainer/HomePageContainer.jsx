import { useMediaQuery, useTheme } from '@mui/material';
import { useParams } from 'react-router-dom';
import { useCookies } from 'react-cookie';

import MainContent from '../../MainContent/MainContent';
import { MainContainer, OuterContainer } from './styles';
import SideBar from '../../SideBar/SideBar';
import Communities from './Communities/Communities';
import RedditPremium from './RedditPremium/RedditPremium';
import PersonalReddit from './PersonalReddit/PersonalReddit';
import HomePageFooter from './HomePageFooter/HomePageFooter';
import PostsClassification from './PostsClassification/PostsClassification';
import CreatePostInHome from './CreatePostInHome/CreatePostInHome';
import BackToTop from '../../BackToTop/BackToTop';
import PostList from './PostList/PostList';
import homePageServer from './homePageServer';
import cleanPage from '../../../utils/cleanPage';

/**
 * This component works as a container for all home page components
 * and a repository the data fetched in
 *
 * @component HomePageContainer
 * @returns {React.Component} Container represents the home page
 */
function HomePageContainer() {
  const { postClass } = useParams();
  // variables
  const theme = useTheme();
  const match = useMediaQuery(theme.breakpoints.up('md'));

  // states
  const [posts, postsError, communities, communitiesError] = homePageServer(postClass);

  // Cookies
  const [cookies] = useCookies(['redditUser']);

  cleanPage();

  return (
    <OuterContainer>
      <MainContainer>
        <MainContent width={640}>
          {cookies.redditUser ? <CreatePostInHome /> : null}
          <PostsClassification />
          {!postsError ? (posts && <PostList posts={posts} />) : 'error in fetching posts'}
        </MainContent>
        {match
        && (
        <SideBar>
          {cookies.redditUser && (!communitiesError ? (communities && <Communities communities={communities} />) : 'error in fetching commmunities')}
          <RedditPremium />
          <PersonalReddit
            title="Home"
            paragraph="Your personal Reddit frontpage. Come here to check in with your favorite communities."
            image="https://www.redditstatic.com/desktop2x/img/id-cards/home-banner@2x.png"
          />
          <HomePageFooter />
          <BackToTop />
        </SideBar>
        )}
      </MainContainer>
    </OuterContainer>
  );
}

export default HomePageContainer;
