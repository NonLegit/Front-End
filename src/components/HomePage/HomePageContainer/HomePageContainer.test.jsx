import renderer from 'react-test-renderer';
import { ThemeProvider } from '@mui/system';
import { MemoryRouter as Router } from 'react-router-dom';
import HomePageContainer from './HomePageContainer';
import theme from '../../../styles/theme';
import ListingContextProvider from '../../../contexts/ListingContext';
import PostTypeContextProvider from '../../../contexts/PostTypeContext';
import HiddenPostsContextProvider from '../../../contexts/HiddenPostsContext';
import EditPostContextProvider from '../../../contexts/EditPostContext';
import CreatePostSidebarContextProvider from '../../../contexts/CreatePostSidebarContext';

describe(HomePageContainer, () => {
  it('PopularContainer renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <CreatePostSidebarContextProvider>
            <ListingContextProvider>
              <PostTypeContextProvider>
                <HiddenPostsContextProvider>
                  <EditPostContextProvider>
                    <ThemeProvider theme={theme}>
                      <HomePageContainer />
                    </ThemeProvider>
                  </EditPostContextProvider>
                </HiddenPostsContextProvider>
              </PostTypeContextProvider>
            </ListingContextProvider>
          </CreatePostSidebarContextProvider>
          ,
        </Router>,

      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
