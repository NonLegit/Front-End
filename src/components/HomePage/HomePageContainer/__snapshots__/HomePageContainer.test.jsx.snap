// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageContainer PopularContainer renders correctly 1`] = `
Array [
  <div
    className="MuiBox-root css-pwymh4"
  >
    <div
      className="css-1yelhn"
    >
      <div
        className="css-3ggxsn"
        width={640}
      >
        <div
          className="css-1i4l6gc"
        />
      </div>
    </div>
  </div>,
  ",",
]
`;
